programs/locker/src/util/token2022.rs

```diff
@@ -0,0 +1,247 @@
+use anchor_lang::prelude::*;
+use anchor_spl::memo;
+use anchor_spl::memo::{BuildMemo, Memo};
+use anchor_spl::token::Token;
+use anchor_spl::token_2022::spl_token_2022::extension::transfer_fee::{
+    TransferFee, TransferFeeConfig, MAX_FEE_BASIS_POINTS,
+};
+use anchor_spl::token_2022::spl_token_2022::{
+    self,
+    extension::{self, StateWithExtensions},
+};
+use anchor_spl::token_interface::spl_token_2022::extension::BaseStateWithExtensions;
+use anchor_spl::token_interface::{Mint, TokenAccount, TokenInterface};
+
+use crate::{LockerError, VestingEscrow};
+
+pub const TRANSFER_MEMO_CLAIM_VESTING: &str = "Jup-Lock ClaimVesting";
+pub const TRANSFER_MEMO_CANCEL_VESTING: &str = "Jup-Lock CancelVesting";
+
+const ONE_IN_BASIS_POINTS: u128 = MAX_FEE_BASIS_POINTS as u128;
+
+#[derive(Clone, Copy)]
+pub struct MemoTransferContext<'a, 'info> {
+    pub memo_program: &'a Program<'info, Memo>,
+    pub memo: &'static [u8],
+}
+
+pub fn transfer_to_escrow_v2<'a, 'c: 'info, 'info>(
+    sender: &'a Signer<'info>,
+    token_mint: &InterfaceAccount<'info, Mint>,
+    sender_token: &InterfaceAccount<'info, TokenAccount>,
+    escrow_token: &InterfaceAccount<'info, TokenAccount>,
+    token_program: &Interface<'info, TokenInterface>,
+    amount: u64,
+) -> Result<()> {
+    let instruction = spl_token_2022::instruction::transfer_checked(
+        token_program.key,
+        &sender_token.key(),
+        &token_mint.key(),   // mint
+        &escrow_token.key(), // to
+        sender.key,          // authority
+        &[],
+        // The transfer amount should include fee
+        amount,
+        token_mint.decimals,
+    )?;
+
+    let account_infos = vec![
+        token_program.to_account_info(),
```

Comment:
Do we need `token_program` in account infos?

Reply:
removed, thank you !